### DISCUSSION TITLE
GPU Gives Different Results with the Same Seed

### DISCUSSION THREAD

**Discussion by pmephin** [glotzerlab/hoomd-blue#1726] [Support] [answered]
Hello, I am a beginner in hoomd-blue and I want to use it for my project. I have installed version 4.5 with an RTX 3050 GPU.

I followed the example code for ‘Initializing a random system’. However, I encountered a problem when I used the GPU as the device. The final compositions were different for each run, even though I used the same seed. This did not happen when I used the CPU. Do you know why this is happening and how to fix it?

**Comment by erjank** [comment, marked as answer]
Thread execution is non-deterministic on the GPU, so you can get different results with the same seed on a GPU. If this is the source of the system differences I don't think you can fix it (but that's probably ok!)

**Reply by joaander** [reply]
Correct. In MD simulations, different thread execution orders will lead to different orders of forces in sums. Due to round-off error, this will lead to slightly different net forces, causing identically initialized MD simulations to diverge exponentially in time. You should use different seeds for replicates and not rely on this behavior.

HOOMD-blue could be modified to implement binary-reproducible simulations (assuming a fixed code binary, execution configuration, and hardware platform) - but at the cost of much slower performance.
